src/RayTracer/Public/js/render-3d.js

```diff
@@ -0,0 +1,161 @@
+const $ = require('jquery');
+
+// standard global variables
+let container, scene, camera, renderer, controls, stats;
+
+// custom global variables
+let targetList = [];
+let projector, mouse = { x: 0, y: 0 };
+	// FUNCTIONS 		
+
+	function rgbToHex(r, g, b) {
+		return "#" + (1 << 24 | r << 16 | g << 8 | b).toString(16).slice(1)
+	}
+
+	exports.init = function (b_color, camera_position, light_position) {
+
+		// SCENE
+		scene = new THREE.Scene();
+		// CAMERA
+		var SCREEN_WIDTH = window.innerWidth, SCREEN_HEIGHT = window.innerHeight;
+		var VIEW_ANGLE = 45, ASPECT = SCREEN_WIDTH / SCREEN_HEIGHT, NEAR = 0.1, FAR = 20000;
+		camera = new THREE.PerspectiveCamera(VIEW_ANGLE, ASPECT, NEAR, FAR);
+		scene.add(camera);
+		camera.position.set(camera_position.x, camera_position.y, camera_position.z);
+		camera.lookAt(scene.position);
+		// RENDERER
+		if (Detector.webgl)
+			renderer = new THREE.WebGLRenderer({ antialias: true });
+		else
+			renderer = new THREE.CanvasRenderer();
+		renderer.setSize(SCREEN_WIDTH, SCREEN_HEIGHT);
+		container = document.getElementById('ThreeJS');
+		container.appendChild(renderer.domElement);
+		// EVENTS
+		THREEx.WindowResize(renderer, camera);
+		THREEx.FullScreen.bindKey({ charCode: 'm'.charCodeAt(0) });
+		// CONTROLS
+		controls = new THREE.OrbitControls(camera, renderer.domElement);
+		// LIGHT
+		var light = new THREE.PointLight(0xffffff);
+		light.position.set(light_position.x, light_position.y, light_position.z);
+		scene.add(light);
+		// FLOOR
+		/* var floorTexture = new THREE.ImageUtils.loadTexture( 'images/checkerboard.jpg' );
+		floorTexture.wrapS = floorTexture.wrapT = THREE.RepeatWrapping; 
+		floorTexture.repeat.set( 10, 10 ); */
```

Comment:
Can we get rid of this commented out code plz

Reply:
The commented code is left there on purpose. This is to do if we want to add textures to the material and the others are to do with the objects being clickable. I have left them just incase we want to implement those features in the future. Ill have everything else fixed today

Reply:
I will leave the commented code in for now.. If in the next weeks, we dont implement these, I will delete them